sdk/core/azure-core/inc/azure/core/http/websockets/curl_websockets_transport.hpp

```diff
@@ -0,0 +1,144 @@
+     * elements.
+     */
+    virtual bool NativeWebsocketSupport() override { return false; }
+
+    virtual void CompleteUpgrade() override;
+
+    /**
+     * @brief Gracefully closes the WebSocket, notifying the remote node of the close reason.
+     *
+     */
+    virtual void Close() override;
+
+    // Native WebSocket support methods.
+    /**
+     * @brief Gracefully closes the WebSocket, notifying the remote node of the close reason.
+     *
+     * @details Not implemented for CURL websockets because CURL does not support native websockets.
+     *
+     */
+    virtual void CloseSocket(uint16_t, std::string const&, Azure::Core::Context const&) override
+    {
+      throw std::runtime_error("Not implemented.");
+    }
+
+    /**
+     * @brief Retrieve the status of the close socket operation.
+     *
+     * @details Not implemented for CURL websockets because CURL does not support native websockets.
+     *
+     */
+    std::pair<uint16_t, std::string> GetCloseSocketInformation(const Azure::Core::Context&) override
+    {
+      throw std::runtime_error("Not implemented");
+    }
+
+    /**
+     * @brief Send a frame of data to the remote node.
+     *
+     * @details Not implemented for CURL websockets because CURL does not support native websockets.
+     *
+     */
+    virtual void SendFrame(
+        WebSocketFrameType,
+        std::vector<uint8_t> const&,
+        Azure::Core::Context const&) override
+    {
+      throw std::runtime_error("Not implemented.");
+    }
+
+    /**
+     * @brief Receive a frame of data from the remote node.
+     *
+     * @details Not implemented for CURL websockets because CURL does not support native websockets.
+     *
+     */
+    virtual std::vector<uint8_t> ReceiveFrame(WebSocketFrameType&, Azure::Core::Context const&)
+        override
+    {
+      throw std::runtime_error("Not implemented");
+    }
+
+    // Non-Native WebSocket support.
+    /**
+     * @brief This function is used when working with streams to pull more data from the wire.
+     * Function will try to keep pulling data from socket until the buffer is all written or until
+     * there is no more data to get from the socket.
+     *
+     * @param buffer Buffer to fill with data.
+     * @param bufferSize Size of buffer.
+     * @param context Context to control the request lifetime.
+     *
+     * @returns Buffer data received.
+     *
+     */
+    virtual size_t ReadFromSocket(uint8_t* buffer, size_t bufferSize, Context const& context)
+        override;
+
+    /**
+     * @brief This method will use libcurl socket to write all the bytes from buffer.
+     *
+     * @param buffer Buffer to send.
+     * @param bufferSize Number of bytes to write.
+     * @param context Context for the operation.
+     */
+    virtual int SendBuffer(uint8_t const* buffer, size_t bufferSize, Context const& context)
+        override;
+    bool SupportsWebSockets() const override { return true; }
```

Comment:
docs

Reply:
Added, but it's the same as the parent method.